tests/rptest/tests/polaris_catalog_smoke_test.py

```diff
@@ -51,3 +63,71 @@ def test_creating_catalog(self):
+
+    @cluster(num_nodes=2)
+    def test_connectiong_to_catalog(self):
+        """The very basic test checking interaction with polaris catalog
+        """
+        polaris_api = PolarisDefaultApi(self.polaris.management_client())
+        catalog_name = "test-catalog"
+        principal_name = "test-user"
+        principal_role_name = "test-user-role"
+        catalog_role_name = "test-catalog-role"
+        catalog = Catalog(
+            type="INTERNAL",
+            name=catalog_name,
+            properties=CatalogProperties(
+                default_base_location=
+                f"file://{PolarisCatalog.PERSISTENT_ROOT}/catalog_data",
+                additional_properties={}),
+            storageConfigInfo=StorageConfigInfo(storageType="FILE"))
+
+        # create a catalog
+        polaris_api.create_catalog(CreateCatalogRequest(catalog=catalog))
+
+        # create principal and grant the catalog role
+        r = polaris_api.create_principal(
+            CreatePrincipalRequest(principal=Principal(name=principal_name)))
+        credentials = r.credentials
+
+        r = polaris_api.create_principal_role(
+            CreatePrincipalRoleRequest(principalRole=PrincipalRole(
+                name=principal_role_name)))
+
+        r = polaris_api.create_catalog_role(
+            catalog_name=catalog_name,
+            create_catalog_role_request=CreateCatalogRoleRequest(
+                catalogRole=CatalogRole(name=catalog_role_name)))
+        for p in [
+                CatalogPrivilege.CATALOG_MANAGE_ACCESS,
+                CatalogPrivilege.CATALOG_MANAGE_CONTENT,
+                CatalogPrivilege.CATALOG_MANAGE_METADATA,
+                CatalogPrivilege.NAMESPACE_CREATE,
+        ]:
+            r = polaris_api.add_grant_to_catalog_role(
+                catalog_name=catalog_name,
+                catalog_role_name=catalog_role_name,
+                add_grant_request=AddGrantRequest(
+                    grant=CatalogGrant(type='catalog', privilege=p)))
+        r = polaris_api.assign_principal_role(
+            principal_name=principal_name,
+            grant_principal_role_request=GrantPrincipalRoleRequest(
+                principalRole=PrincipalRole(name=principal_role_name)))
+
+        r = polaris_api.assign_catalog_role_to_principal_role(
+            catalog_name=catalog_name,
+            principal_role_name=principal_role_name,
+            grant_catalog_role_request=GrantCatalogRoleRequest(
+                catalogRole=CatalogRole(name=catalog_role_name)))
+
+        py_catalog = load_catalog(
+            "test-catalog", **{
+                "uri": f"{self.polaris.catalog_url}",
+                'type': "rest",
+                'scope': "PRINCIPAL_ROLE:ALL",
+                "credential":
+                f"{credentials.client_id}:{credentials.client_secret}",
+                "warehouse": "test-catalog",
+            })
+
+        py_catalog.create_namespace_if_not_exists("test_namespace")
```

Comment:
nit: since `create_namespace_if_not_exists` does not return a result, should there be a `list_namespaces` after this to confirm the call succeeded?